import-scripts/import-devdb-data.sh

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
+
+FLOCK_FILEPATH="/data/portal-cron/cron-lock/import-devdb-data.lock"
+(
+    # check lock so that script executions do not overlap
+    if ! flock --nonblock --exclusive $flock_fd ; then
+        exit 0
+    fi
+
+    # set necessary env variables with automation-environment.sh
+    if [ -z "$PORTAL_HOME" ] || [ -z "$START_DEVDB_IMPORT_TRIGGER_FILENAME" ] || [ -z "$KILL_DEVDB_IMPORT_TRIGGER_FILENAME" ] || [ -z "$DEVDB_IMPORT_IN_PROGRESS_FILENAME" ] || [ -z "$DEVDB_IMPORT_KILLING_FILENAME" ] ; then
+        echo "Error : import-devdb-data.sh cannot be run without setting the PORTAL_HOME, START_DEVDB_IMPORT_TRIGGER_FILENAME, KILL_DEVDB_IMPORT_TRIGGER_FILENAME, DEVDB_IMPORT_IN_PROGRESS_FILENAME, DEVDB_IMPORT_KILLING_FILENAME environment variables. (Use automation-environment.sh)"
+        exit 1
+    fi
+
+    if [ -f "$KILL_DEVDB_IMPORT_TRIGGER_FILENAME" ] || [ -f "$DEVDB_IMPORT_KILLING_FILENAME" ] ; then
+        # if import kill is happening or is triggered, cancel any import trigger and exit
+        rm -f "$START_DEVDB_IMPORT_TRIGGER_FILENAME"
+        exit 1
+    fi
+
+    if ! [ -f "$START_DEVDB_IMPORT_TRIGGER_FILENAME" ] ; then
+        # no start trigger for import, so exit
+        exit 1
+    fi
+
+    echo $(date)
+
+    # set data source env variables
+    source $PORTAL_HOME/scripts/set-data-source-environment-vars.sh
+    source $PORTAL_HOME/scripts/clear-persistence-cache-shell-functions.sh
+
+    tmp=$PORTAL_HOME/tmp/import-cron-devdb
+    if ! [ -d "$tmp" ] ; then
+        if ! mkdir -p "$tmp" ; then
+            echo "Error : could not create tmp directory '$tmp'" >&2
+            exit 1
+        fi
+    fi
+    if [[ -d "$tmp" && "$tmp" != "/" ]]; then
+        rm -rf "$tmp"/*
+    fi
+    now=$(date "+%Y-%m-%d-%H-%M-%S")
+    IMPORTER_JAR_FILENAME="$PORTAL_HOME/lib/devdb-importer.jar"
+    java_debug_args=""
+    ENABLE_DEBUGGING=0
+    if [ $ENABLE_DEBUGGING != "0" ] ; then
+        java_debug_args="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=27183"
+    fi
+    JAVA_IMPORTER_ARGS="$JAVA_PROXY_ARGS $java_debug_args $JAVA_SSL_ARGS -Dspring.profiles.active=dbcp -Djava.io.tmpdir=$tmp -ea -cp $IMPORTER_JAR_FILENAME org.mskcc.cbio.importer.Admin"
+    devdb_notification_file=$(mktemp $tmp/devdb-portal-update-notification.$now.XXXXXX)
+    ONCOTREE_VERSION_TO_USE=oncotree_candidate_release
+    DATA_SOURCES_TO_BE_FETCHED="datahub impact"
```

Comment:
will need to include whatever new forks we're using to mimic private repos - I can add this once I iron this out with Anusha/Ramya

Reply:
👍  There are some notes in our scrum channel about other things which might need to be tweaked (jar filename, log filename, importer configuration sheet header)